### DISCUSSION TITLE
Hash router

### DISCUSSION THREAD

**Discussion by PointSingularity** [zoontek/chicane#40] [Q&A] [answered]
Hi, I am really liking how Chicane looks!

I was wondering is there a way to achieve [Hash Routing](https://reactrouter.com/en/main/router-components/hash-router)?

For some of my projects, I don't have full control of the environment, so having the routing be fully client side, is a must.

**Comment by zoontek** [comment, marked as answer]
Hi, thanks for your interest in Chicane,

Hash routing is very limited since it's not compatible with server-side rendering, as hash is not sent to the backend. We do not plan to add this, but feel free to fork and implement it.

Note that chicane can be used 100% on the client, it uses [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API) (similarly as RR [BrowserRouter](https://reactrouter.com/en/main/router-components/browser-router)). Hash routing is only needed for extra old browsers or if you are not allowed to use the location path for some reason, this is not related to your backend environment at all.

**Reply by PointSingularity** [reply]
> Hi, thanks for your interest in Chicane,
> 
> Hash routing is very limited since it's not compatible with server-side rendering, as hash is not sent to the backend. We do not plan to add this, but feel free to fork and implement it.
> 
> Note that chicane can be used 100% on the client, it uses [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API) (similarly as RR [BrowserRouter](https://reactrouter.com/en/main/router-components/browser-router)). Hash routing is only needed for extra old browsers or if you are not allowed to use the location path for some reason, this is not related to your backend environment at all.

Thanks for your answer!

I think I wasn't clear enough, basically I don't have control over the redirects on some production deployments, which makes things unnecessarily tricky. That's why I've been using the hashing strategy, because it circumvents that.

But I do understand that is not the way to go.

If I find some free time, I'll try to implement it in a fork 😄
